infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -119,21 +105,34 @@ private void openTables() {
     public Set<String> getStreamsToReplicate() {
         try {
             currentVersion = logReplicationConfigAdapter.getVersion();
-            Set<String> fetchedStreams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+            Set<String> streams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+            // Add registryTable to the streams
+            String registryTable = getFullyQualifiedTableName(
+                    CORFU_SYSTEM_NAMESPACE, TableRegistry.REGISTRY_TABLE_NAME);
+            streams.add(registryTable);
+
+            // Add protoBufDescriptorTable to the streams
+            String protoTable = getFullyQualifiedTableName(
+                    CORFU_SYSTEM_NAMESPACE, TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME);
+            streams.add(protoTable);
 
             return IRetry.build(IntervalRetry.class, () -> {
-                Set<String> streams = new HashSet<>();
                 try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
                     VersionResult result = verifyVersionResult(txn);
                     if (result.equals(VersionResult.UNSET) || result.equals(VersionResult.CHANGE)) {
+                        // Case of upgrade or initial boot: sync version table with plugin info
```

Comment:
Shouldn't we check if the old STREAMS_TO_REPLICATE table exists/is present in the registry and if so remove it from the registry (so it is no longer checkpointed/trimmed) otherwise it will remain over time (no actual cleanup) after this change is introduced.

Reply:
I got the point, but it looks like we will add this remove STREAMS_TO_REPLICATE  table workflow just for executing 1 time?

Reply:
Correct, it would execute one time but it would safeguard any upgrade to this new version... or any future version if you jump between versions. Anyways the version upgrade is not constantly happening.